src/plugins/codeStyle/index.ts

```diff
@@ -18,13 +36,46 @@ export default class CodeStyle {
         extractFixes(addFixes, event.diagnostics);
     }
 
-    afterFileValidate(file: BscFile) {
-        if (!isBrsFile(file) || this.lintContext.ignores(file)) {
-            return;
+    validateXMLFile(file: XmlFile) {
+        const diagnostics: Omit<BsDiagnostic, "file">[] = [];
+        const { noArrayComponentFieldType, noAssocarrayComponentFieldType } = this.lintContext.severity;
+
+        const validateArrayComponentFieldType = noArrayComponentFieldType !== DiagnosticSeverity.Hint;
+        const validateAssocarrayComponentFieldType = noAssocarrayComponentFieldType !== DiagnosticSeverity.Hint;
+
+        for (const field of file.parser?.ast?.component?.api?.fields ?? []) {
+            const { tag, attributes, range } = field;
+            if (tag.text === "field") {
+                const typeAttribute = attributes.find(
+                    ({ key }) =>
+                        key.text === "type"
+                );
+
+                const typeValue = typeAttribute?.value.text;
+                if (typeValue === 'array' && validateArrayComponentFieldType) {
+                    diagnostics.push(
+                        messages.noArrayFieldType(
+                            range,
```

Comment:
Sorry, one more thing I just noticed from the original screenshot. Can you update this range to only be the text inside the attribute quotes? For example:

```xml
<field id="something" type="array" />
                            ~~~~~
```

Reply:
Range has been adjusted